### PR TITLE
Include index list from undet of complex lanes

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

**Comment by chuan-wang** [comment]
Tested and worked as expected. /CW

**Review by ssjunnebo — APPROVED** [review]
Great work! 👏 I only had two minor comments, but I'm happy to merge when you are 🙂

File: taca/illumina/Runs.py [hunk 1]
```diff
@@ -431,10 +431,45 @@ def _rename_undet(self, lane, samples_per_lane):
...
+            demux_id = os.path.splitext(os.path.split(samplesheet)[1])[0].split("_")[1]
+            ssparser = SampleSheetParser(samplesheet)
+            for row in ssparser.data:
+                if row['Lane'] not in lane_demuxid_indexlength.keys():
+                    lane_demuxid_indexlength[row['Lane']] = {demux_id: [len(row.get('index','')), len(row.get('index2',''))]}
+                else:
+                    if demux_id not in lane_demuxid_indexlength[row['Lane']].keys():
+                        lane_demuxid_indexlength[row['Lane']][demux_id] = [len(row.get('index','')), len(row.get('index2',''))]
+                    else:
+                        pass
```
[the remark this answers is not in the document]
Comment (chuan-wang) [reply taca/illumina/Runs.py]: Fixed as suggested. /CW

File: taca/illumina/Runs.py [hunk 2]
```diff
@@ -607,21 +681,18 @@ def _aggregate_demux_results_simple_complex(self, simple_lanes, complex_lanes):
...
                         #I update only the importat fields
                         lanes_present_in_stats_json = [entry['LaneNumber'] for entry in stats_list['ConversionResults']]
                         for ReadInfosForLanes_lane in data['ReadInfosForLanes']:
                             if ReadInfosForLanes_lane['LaneNumber'] not in lanes_present_in_stats_json:
                                 stats_list['ReadInfosForLanes'].extend([ReadInfosForLanes_lane])
-                        for ConversionResults_lane  in data['ConversionResults']:
-                            if ConversionResults_lane['LaneNumber'] in lanes_present_in_stats_json:
+                        for ConversionResults_lane in data['ConversionResults']:
+                            if ConversionResults_lane['LaneNumber'] in lanes_present_in_stats_json and str(ConversionResults_lane['LaneNumber']) in complex_lanes.keys():
                                 #i have found the same lane, all these things do not make sense because I have demuxed the lane twice
```
[the remark this answers is not in the document]
Comment (chuan-wang) [reply taca/illumina/Runs.py]: Fixed. /CW

**Pull request merged**